app/controllers/observations_controller/new_and_create.rb

```diff
@@ -102,6 +112,7 @@ def create
     success = false if @name && !@vote.value.nil? && !validate_object(@vote)
     success = false if @bad_images != []
     success = false if success && !save_observation(@observation)
+    update_field_slip(@observation, params[:field_code]) if success
```

Comment:
Suggestions:
1. Put this line with the other things that get executed `if success`. (Result: one fewer conditional.)
2. Consider a guard clase.

The net result would be something like this:
```ruby
    success = false if success && !save_observation(@observation)
    return reload_new_form(params.dig(:naming, :reasons)) unless success

    # Once observation is saved we can save everything else.
    @observation.log(:log_observation_created)
    update_field_slip(@observation, params[:field_code])
    ...
    redirect_to_next_page
```

Reply:
Nice!